### PR TITLE
Add support for Kafka 2.7.0

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

This is a draft PR for Kafka 2.7.0 support. It is currently based on the RC5 release.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=upgrade

**Comment by scholzj** [comment]
@strimzi/maintainers Since the 2.7.0 artifacts are out now, this should be ow ready for review! So please have a look. Thanks

File: helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl
Comment (ppatierno) [inline helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl:21]: shouldn't we delete above 2.5.x? The same applies to the other env vars.

Comment (scholzj) [reply helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl:21]: 2.5.0 is still supported. I do not see any reason for removing it since so far there are no features implemented which would not support 2.5.0 / 2.5.1.

Comment (ppatierno) [reply helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl:21]: it confuses me because in the upgrade doc we have a lower Kafka version as 2.6 so people coming from 2.5.x could be confused as well?

Comment (scholzj) [reply helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl:21]: If the upgrade docs talk about upgrade from 2.5 then the users upgrading from 2.6 will be confused. I think using the latest supported Kafka from previous version makes more sense in this context.

Comment (ppatierno) [reply helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl:21]: Anyway it's going to be confusing :)

**Review by ppatierno — APPROVED** [review]

File: helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl
Comment (scholzj) [reply helm-charts/helm2/strimzi-kafka-operator/templates/_kafka_image_map.tpl:21]: It will ... but I think that keeping 2.5 is still more user-friendly then removing it to he the upgrade docs fit in 100% of cases. Thanks for approving.

**Comment by syepes** [comment]
@scholzj
Thanks for this update, is there any ETA on when this will be included in the official release?

**Comment by scholzj** [comment]
Kafka released too late before Christmas, so we did not want to have a release in progress during the holidays. I guess now we start releasing after we merge some of the outstanding PRs which accumulated since since they have some new bug-fixes / enhancements.

**Comment by syepes** [comment]
Ok thanks for the info, I'll keep an eye :-)

**Pull request merged**
